### PR TITLE
Remove trailing slash when passing directory path to GetContent

### LINKED ISSUE #1629: Remove trailing slashes from file "paths" to build correct urls
Hello,
I try to download a file from a different branch than `master`.

Request to:
```
https://api.github.com/repos/StarpTech/shikaka/contents/.github/workflows/?ref=feature%2Fghconfig_update_workflows
```

Response:

```json
[
    {
        "name": "nodejs.yml",
        "path": ".github/workflows/nodejs.yml",
        "sha": "cb4ce2e0cee2cf5af199fc69610b7442e9118b2b",
        "size": 709,
        "url": "https://api.github.com/repos/StarpTech/shikaka/contents/.github/workflows/nodejs.yml?ref=master",
        "html_url": "https://github.com/StarpTech/shikaka/blob/master/.github/workflows/nodejs.yml",
        "git_url": "https://api.github.com/repos/StarpTech/shikaka/git/blobs/cb4ce2e0cee2cf5af199fc69610b7442e9118b2b",
        "download_url": "https://raw.githubusercontent.com/StarpTech/shikaka/master/.github/workflows/nodejs.yml",
        "type": "file",
        "_links": {
            "self": "https://api.github.com/repos/StarpTech/shikaka/contents/.github/workflows/nodejs.yml?ref=master",
            "git": "https://api.github.com/repos/StarpTech/shikaka/git/blobs/cb4ce2e0cee2cf5af199fc69610b7442e9118b2b",
            "html": "https://github.com/StarpTech/shikaka/blob/master/.github/workflows/nodejs.yml"
        }
    }
]
```

returns always file references to master and it doesn't matter what `ref` I pass.

This looks like a Github API issue. I added it here due to transparency reasons and to improve the integration test suite.

### PR DESCRIPTION
**Pull request by StarpTech**
Fix #1629 

To avoid producing an invalid request URL like `.github/?ref`.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @StarpTech !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
Thank you, @StarpTech 
LGTM 👌🏼

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
